mods/default/functions.lua

```diff
@@ -481,3 +481,35 @@ minetest.register_abm({
 		end
 	end
 })
+
+
+--
+-- Checks if specified volume intersects a protected volume
+--
+
+-- Checks corners first then interior volume at specified intervals
+
+function default.intersects_protection(minp, maxp, name, interval)
+	if minetest.is_protected({x = minp.x, y = minp.y, z = minp.z}, name) or
+			minetest.is_protected({x = minp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = minp.x, y = maxp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = minp.y, z = maxp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = minp.z}, name) or
+			minetest.is_protected({x = maxp.x, y = maxp.y, z = maxp.z}, name) then
+		return true
+	end
+
+	for z = minp.z + interval, maxp.z - 1, interval do
+	for y = minp.y + interval, maxp.y - 1, interval do
+	for x = minp.x + interval, maxp.x - 1, interval do
+		if minetest.is_protected({x = x, y = y, z = z}, name) then
```

Comment:
I think these loops do not visit the borders of the area, other than at the corners. E.g. `{ x = minp.x + 5, y = minp.y, z = minp.z + 5}` will not be vistited. This would
allow placing a sapling below a protected area near to its border, for a tree which will extend to above the protected area. As the borders of the tree-area are only checked exactly above and below the protected area, and not in the middle of the area, this would allow leaves in the middle to grief the protected area.

Example: a sapling for the following tree would be placeable:
```
LLLLLLLLLLL
LLLLLL|---------------------
LLLLLL|LLLL
LLLLLL|LLLL
LLLLLL|LLLL
LLLLLL|LLLL
LLLLLL|LLLL   protected
     T|        area
     T|
     T|
     T|-----------------------
     T
```